lib/api/apiUtils/object/corsResponse.js

```diff
@@ -0,0 +1,127 @@
+/** _matchesValue - compare two values to determine if they match
+* @param {string} allowedValue - an allowed value in a CORS rule;
+    may contain wildcards
+* @param {string} value - value from CORS request
+* @return {boolean} - true/false
+*/
+function _matchesValue(allowedValue, value) {
+    const wildcardIndex = allowedValue.indexOf('*');
+    // If no wildcards, simply return whether strings are equal
+    if (wildcardIndex === -1) {
+        return allowedValue === value;
+    }
+    // Otherwise make sure substrings match expected substrings before
+    // and after the wildcard
+    const beginValue = allowedValue.substring(0, wildcardIndex);
+    const endValue = allowedValue.substring(wildcardIndex + 1);
+    return (value.startsWith(beginValue) && value.endsWith(endValue));
+}
+
+/** _matchesOneOf - check if a value matches any values of a CORS rule array
+* @param {string[]} allowedValues - array of values allowed in CORS rule
+* @param {string} value - value from CORS request
+* @return {boolean} - true/false
+*/
+function _matchesOneOf(allowedValues, value) {
+    return allowedValues.some(allowedValue =>
+        _matchesValue(allowedValue, value));
+}
+
+/** _headersMatchRule - check if headers match AllowedHeaders of rule
+* @param {string[]} headers - expected headers of CORS request
```

Comment:
how about -- the value of the 'Access-Control-Allow-Headers' in an options request and the actual request header keys in any other request'